test/components/connect.spec.js

```diff
@@ -856,6 +856,77 @@ describe('React', () => {
       })
     })
 
+    it('should not set refreshTimeouts when component is unmounted', (done) => {
+      const interval = 100000 // set sufficiently far out to not happen during test
+
+      @connect(() => ({ testFetch: { url: `/example`, refreshInterval: interval } }))
+      class WithProps extends Component {
+        render() {
+          return <Passthrough {...this.props}/>
+        }
+      }
+
+      class OuterComponent extends Component {
+        constructor() {
+          super()
+          this.state = { render: true }
+        }
+
+        setRender(render) {
+          this.setState({ render })
+        }
+
+        render() {
+          if (this.state.render) {
+            return (
+              <div>
+                <WithProps {...this.state} />
+              </div>
+            )
+          } else {
+            return <div />
+          }
+        }
+      }
+
+      const outerComponent = TestUtils.renderIntoDocument(
+        <OuterComponent />
+      )
+
+      const container = TestUtils.findRenderedComponentWithType(outerComponent, WithProps)
+
+      expect(container.state.data.testFetch).toIncludeKeyValues({
+        fulfilled: false, pending: true, reason: null, refreshing: false, rejected: false, settled: false, value: null
+      })
+      expect(container.state.mappings.testFetch.refreshInterval).toEqual(interval)
+      expect(container.state.refreshTimeouts.testFetch).toEqual(undefined)
+
+      setImmediate(() => {
+        expect(container.state.data.testFetch).toIncludeKeyValues({
+          fulfilled: true, pending: false, reason: null, refreshing: false, rejected: false, settled: true, value: { T: 't' }
+        })
+        expect(container.state.mappings.testFetch.refreshInterval).toEqual(interval)
+        const refreshTimeout = container.state.refreshTimeouts.testFetch
+        expect(refreshTimeout).toBeTruthy()
+        const after = refreshTimeout._onTimeout
+
+        // Cancel scheduled refresh
+        clearTimeout(refreshTimeout)
+
+        outerComponent.setRender(false)
+
+        // force the refresh to happen now after it has been unmounted
+        after()
+
+        const spy = expect.spyOn(window, 'setTimeout')
```

Comment:
Hmm.. You're right that it is strange that it is after but I was following the `should not call setState if component is unmounted'` test which does it similarly. But it seems to be working like this because it fails without my change and passes with it.